db/version_set.h

```diff
@@ -657,6 +686,13 @@ class VersionStorageInfo {
   int l0_delay_trigger_count_ = 0;  // Count used to trigger slow down and stop
                                     // for number of L0 files.
 
+  // Compact cursors for round-robin compactions in each level
+  std::vector<InternalKey> compact_cursor_;
+  // The current compact file index in a round-robin manner. We update the
+  // current file to be compacted in `UpdateFilesByCompactionPri(...)` instead
+  // of the current compact cursor.
+  std::vector<int> current_file_to_cursor_;
```

Comment:
I think the complexity to keep these two in sync is nontrivial. Can we use `compact_cursor_` only? It can be used directly to pick a file in `PickCompaction()` and updated in the same function.

Reply:
I have removed the extra vector in a newer version. I am thinking about how to pick a file in`PickCompaction()` and update it.

Reply:
> I have removed the extra vector in a newer version. I am thinking about how to pick a file in`PickCompaction()` and update it.

Not sure if you're still thinking about using cursor directly in `PickCompaction()`. I tried it actually and ended up preferring your current approach. My attempt is in this code (ignore the `AdvanceStartLevelCursor()` - I wrote that before we decided to update cursor elsewhere) - https://github.com/ajkr/rocksdb/commit/f407819863da8835720f5a572e2c60178c9b175e. In the end it looks cleaner to use the `UpdateFilesByCompactionPri()`. Feel free to resolve this.